DG: Update target user profile, value proposition, user stories, glos… (#39)
AY2425S1-CS2103T-F08-2/tp: docs/DeveloperGuide.md

**Target user profile**:

Researchers who..
* have a need to manage a significant number of study participants

> * need an easy way to contact different experimental groups

Reviewer: This doesn't seem to be covered by our intended feature(s). Do you think we should remove it from the user profile, or at least modify it a little to better fit the vision for the project?

Reviewer 2: hmmm i feel like it is covered by our features tbh. cos u can filter list of contacts by study group then export that list of contacts (i think?)

Reviewer: Yup I think you might be right. While our feature doesn't cover the entire point, it does facilitate part of it. Let's keep it for now. 

Thank you @PrishaVP

* need quick access to study participants' information
* prefer desktop apps over other types
* can type fast
* prefers typing to mouse interactions

> * is reasonably comfortable using CLI apps

Reviewer 3: May italicize `CLI` to indicate that its definition is in our glossary!

> * is reasonably comfortable using CLI apps

Reviewer 2: ```suggestion
* is reasonably comfortable using _CLI_ apps
```

> **Value proposition**:

Reviewer 2: ```suggestion
**Value proposition**:

```

As researchers have to handle large groups of participants across multiple studies, *ResearchRoster* allows them to 

> * have all participants consolidated in a single program. 

Reviewer 2: ```suggestion
* have all participants consolidated in a single program,
```

> * consolidate a list of contact details based on **specific experimental criteria**

Reviewer 2: ```suggestion
* consolidate a list of contact details based on **specific experimental criteria**,
```

> * export it to an easy-to-read format for better data organization.

Reviewer 3: May omit the full-stops for consistency!

> Thus, *ResearchRoster* allows researchers to save time, effort and energy whilst keeping their participant data well-organised.

Reviewer 3: ```suggestion
Thus, `ResearchRoster` allows researchers to save time, effort and energy whilst keeping their participant data well-organised.
```

### User stories

Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unlikely to have) - `*`

> | Priority | As a …​    | I want to …​                      | So that I can…​                            |

Reviewer 2: ```suggestion
| Priority | As a …​    | I want to …​                      | So that I can …​                           |
```

| -------- |------------|-----------------------------------|--------------------------------------------|

> | `* * *`  | user       | add a new persson                 | store a new person in my contact list      |

Reviewer 2: ```suggestion
| `* * *`  | user       | add a new participant                 | store a new participant in my contact list      |
```

> | `* * *`  | user       | add a new persson                 | store a new person in my contact list      |

Reviewer 2: ```suggestion
| `* * *`  | user       | add a new person                 | store a new person in my contact list      |
```

**Target user profile**:

Researchers who,
* have a need to manage a significant number of study participants
* need an easy way to contact different experimental groups
* need quick access to study participants' information
* prefer desktop apps over other types
* can type fast
* prefers typing to mouse interactions
* is reasonably comfortable using *CLI* apps

**Value proposition**:

As researchers have to handle large groups of participants across multiple studies, `ResearchRoster` allows them to 
* have all participants consolidated in a single program 
* consolidate a list of contact details based on **specific experimental criteria**
* export it to an easy-to-read format for better data organization

> Thus, *ResearchRoster* allows researchers to save time, effort and energy whilst keeping their participant data well-organised

Reviewer 3: ```suggestion
Thus, *ResearchRoster* allows researchers to save time, effort and energy whilst keeping their participant data well-organised.
```

### Glossary

* **Mainstream OS**: Windows, Linux, Unix, MacOS

> * **GUI**: Graphical User Interface - A visual-based interface where users interact with the system through graphical elements like icons and windows.

Reviewer 2: ```suggestion
* **GUI**: Graphical User Interface - A visual-based interface where users interact with the system through graphical elements like icons and windows
```

> * **CLI**: Command Line interface - A text-based interface where users interact with the system by typing commands.

Reviewer 2: ```suggestion
* **CLI**: Command Line Interface - A text-based interface where users interact with the system by typing commands
```

### Glossary

* **Mainstream OS**: Windows, Linux, Unix, MacOS
* **Private contact detail**: A contact detail that is not meant to be shared with others

> * **UI**: User Interface: The medium through which users interact with a system, encompassing both graphical *(GUI)* and text-based *(CLI)* elements.

Reviewer 3: ```
*(GUI)* -> (*GUI*)
*(CLI)* -> (*CLI*)
```
And can also omit the full-stop!

* **Mainstream OS**: Windows, Linux, Unix, MacOS
* **Private contact detail**: A contact detail that is not meant to be shared with others

> * **UI**: User Interface: The medium through which users interact with a system, encompassing both graphical *(GUI)* and text-based *(CLI)* elements

Reviewer 2: ```suggestion
* **UI**: User Interface: The medium through which users interact with a system, encompassing both graphical (*GUI*) and text-based (*CLI*) elements
```

> * **UI**: User Interface: The medium through which users interact with a system, encompassing both graphical (*GUI*) and text-based (*CLI*) elements

Reviewer 3: ```suggestion
* **UI**: User Interface - The medium through which users interact with a system, encompassing both graphical (*GUI*) and text-based (*CLI*) elements
```

> * **UI**: User Interface: The medium through which users interact with a system, encompassing both graphical *(GUI)* and text-based *(CLI)* elements.

Reviewer 2: ```suggestion
* **UI**: User Interface: The medium through which users interact with a system, encompassing both graphical *(GUI)* and text-based *(CLI)* elements
```
